.github/workflows/mdbook.yml

```diff
@@ -21,29 +21,29 @@ concurrency:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
-    env:
-      MDBOOK_VERSION: "0.4.36"
+    runs-on: nixos-24.05
+
     steps:
-      - name: install
-        run: |
-          set -x
-          wget -O - \
-            "https://github.com/rust-lang/mdBook/releases/download/v${MDBOOK_VERSION}/mdbook-v${MDBOOK_VERSION}-x86_64-unknown-linux-gnu.tar.gz" \
-          | sudo tar xzf - -C /usr/local/bin
-      - name: checkout
-        uses: actions/checkout@v3
-        with:
-          submodules: recursive
-      - name: build
-        run: |
-          mdbook build
-      - name: configure
-        uses: actions/configure-pages@v3
-      - name: upload
-        uses: actions/upload-pages-artifact@v2
-        with:
-          path: ./book/
+    - uses: actions/checkout@v4
+
+    - name: Install Nix
+      uses: cachix/install-nix-action@v27
+      with:
+        extra_nix_config: |
+          substituters = https://nix-cache.lowrisc.org/public/ https://cache.nixos.org/
+          trusted-public-keys = nix-cache.lowrisc.org-public-1:O6JLD0yXzaJDPiQW1meVu32JIDViuaPtGDfjlOopU7o= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
+
+    - name: Build Documentation
+      run: |
+        nix build .#sonata-documentation
+        # copy out of /nix/store with all symlinks dereferenced and with default permissions
+        cp -LR --no-preserve=mode,ownership result build
```

Comment:
You probably also want to ignore timestamps, although I am not sure if GitHub pages action actually care or not.

Reply:
Removed ownership